Add more compiler version info to INSTALL.rst (#1028)
dftbplus/dftbplus: INSTALL.rst

* Python (version >= 3.2) for the source preprocessor

Fortran compiler
----------------

> The following Fortran compilers are known to build DFTB+:

Reviewer: ```suggestion
The following Fortran compilers are known to build DFTB+ correctly:
```

* GNU >= 9

> * Intel >= 19 or >= 2020 (when built with the tblite library, tblite is known to

Reviewer: ```suggestion
* Intel >= 19 or >= 2020 (when built to include the tblite library, as tblite is known to
```

  produce incorrect results with Intel 19)

* NAG >= 7.1 (when built without OpenMP support)

The following Fortran compilers are known to fail to build DFTB+:

* NAG (when built with OpenMP support, unsupported OpenMP 4.0 constructs, last
  tested version: 7.1)

* NVIDIA (internal compiler error & unsupported OpenMP 4.0 constructs, last
  tested version: 22.3)

Older versions of the compilers above are likely to fail due to missing Fortran

> features and/or compiler bugs, please use the versions listed above. Compilers

Reviewer: ```suggestion
features and/or compiler bugs. Compilers
```
Think the "please use" part is implied.

> by other vendors may work, but had not been tested extensively (see also

Reviewer: ```suggestion
by other vendors may work, but have not been tested extensively (see also
```

> `Tested build environments <#tested-build-environments>`_).

Reviewer: ```suggestion
`Tested build environments <#tested-build-environments>`_ and `Testing DFTB+ <#testing-dftb+>`_).
```
(haven't tested the suggested link in place)
